x-pack/plugins/alerts/server/task_runner/task_runner.ts

```diff
@@ -91,9 +92,10 @@ export class TaskRunner {
       requestHeaders.authorization = `ApiKey ${apiKey}`;
     }
 
-    return ({
+    const path = spaceId ? `/s/${spaceId}` : '/';
+
+    const fakeRequest = KibanaRequest.from(({
       headers: requestHeaders,
-      getBasePath: () => this.context.getBasePath(spaceId),
```

Comment:
This change looks wrong to me... but I'm not actually sure.
I thought `context.getBasePath` was implemented by the Spaces plugin and so line 95 feels like Spaces domain... 🤔 
In plugin.ts we do this:
```
  private getBasePath = (spaceId?: string): string => {
    return this.spaces && spaceId ? this.spaces.getBasePath(spaceId) : this.serverBasePath!;
  };
```

Perhaps @mikecote can help here?

Reply:
> I thought context.getBasePath was implemented by the Spaces plugin and so line 95 feels like Spaces domain...

Yeah we were using `getBasePath` from the Spaces plugin in order to simulate a request coming from the Legacy platform. This PR removed support for the LP requests, and instead forces consumers to construct `KibanaRequest` instances. Alerting & Actions were creating a LP fake request, but casting it as a `KibanaRequest`.

One of the reasons to fake a request was so that the Spaces plugin could properly scope instances of the Saved Objects client. Since we aren't honoring LP requests anymore, there isn't a need to provide the `getBasePath` function -- rather, consumers need to inform core about the request's base path by calling `core.http.basePath.set(fakeRequest, spaceAwarePath)` instead.

^^^^^
At least, this is the intent behind my most recent set of changes.

Reply:
@mikecote what are your thoughts?

Reply:
> consumers need to inform core about the request's base path by calling `core.http.basePath.set(fakeRequest, spaceAwarePath)` instead.

If that is their recommendation for now, I'm ok with this change. Alerting is hacking fake requests in Kibana until [scope-able elasticsearch clients](https://github.com/elastic/kibana/issues/39430) is around so we'll have some odd code to deal with until then either way.

Reply:
In that case my review still stands, but on this specific question I'm relying on your understanding of this 😆